sale_promotion_discount_in_field/wizard/sale_coupon_apply_code.py

```diff
@@ -0,0 +1,27 @@
+from odoo import models
+from odoo.osv import expression
+
+
+class SaleCouponApplyCode(models.TransientModel):
+    _inherit = "sale.coupon.apply.code"
+
+    def apply_coupon(self, order, coupon_code):
+        # OVERRIDE to apply the program to the order without reward lines
+        error_status = {}
+        program_domain = order._get_coupon_program_domain()
+        program_domain = expression.AND(
+            [program_domain, [("promo_code", "=", coupon_code)]]
+        )
+        program = self.env["coupon.program"].search(program_domain)
+        if (
+            program
+            and program.reward_type == "discount_line"
```

Comment:
I think this conditions need add in search domain on the 15th code lines.
Then like this:
```
if program:
   ...
   return error_status
return super().apply_coupon(order, coupon_code)
```

Reply:
resolved